### PR TITLE
bpo-42158: Add MIME types for n-triples, n-quads, n3 and trig

### PR DESCRIPTION
**Pull request by DylanVanAssche**
Adds MIME types to `Lib/mimetypes.py` for:
- N-quads: `.nq: application/n-quads` (specification: https://www.w3.org/TR/n-quads/)
- N-triples: `.nt: application/n-triples` (specification: https://www.w3.org/TR/n-triples/)
- Notation3: `.n3: text/n3` (specification: https://www.w3.org/TeamSubmission/n3/)
- Trig: `.trig: application/trig` (specification: https://www.w3.org/TR/trig/)

https://bugs.python.org/issue42158

### CODE REVIEW

**Comment by the-knights-who-say-ni** [comment]
Hello, and thanks for your contribution!

I'm a bot set up to make sure that the project can legally accept this contribution by verifying everyone involved has signed the [PSF contributor agreement](https://www.python.org/psf/contrib/contrib-form/) (CLA).



## CLA Missing

Our records indicate the following people have not signed the CLA:

@DylanVanAssche

For legal reasons we need all the people listed to sign the CLA before we can look at your contribution. Please follow [the steps outlined in the CPython devguide](https://devguide.python.org/pullrequest/#licensing) to rectify this issue.

If you have recently signed the CLA, please wait at least one business day
before our records are updated.


You can [check yourself](https://check-python-cla.herokuapp.com/) to see if the CLA has been received.

Thanks again for the contribution, we look forward to reviewing it!

**Comment by pheyvaer** [comment]
This would be very useful my work too!

File: Misc/NEWS.d/next/Library/2020-11-10-17-46-12.bpo-42158.OhxAiH.rst [hunk 1]
```diff
@@ -0,0 +1 @@
+Add MIME types for N-quads, N-triples, Notation3 and Trix to `Lib/mimetypes.py`
```
Comment (merwok) [inline Misc/NEWS.d/next/Library/2020-11-10-17-46-12.bpo-42158.OhxAiH.rst]:
```suggestion
Add MIME types for N-quads, N-triples, Notation3 and TriG to ``mimetypes``.
```

**Review by merwok — APPROVED** [review]
I can confirm these 4 entries exist in /etc/mime.types in current Debian testing.

**Comment by DylanVanAssche** [comment]
> I can confirm these 4 entries exist in /etc/mime.types in current Debian testing.

Not in Ubuntu 20.04 LTS unfortunately.

**Comment by merwok** [comment]
Sorry for the confusion — I wasn’t saying that to mean you should not change Python’s mimetypes, but to mean that the mappings are correct 🙂

**Review by merwok — APPROVED** [review]

**Comment by DylanVanAssche** [comment]
> Sorry for the confusion — I wasn’t saying that to mean you should not change Python’s mimetypes, but to mean that the mappings are correct slightly_smiling_face

No problem :smiley:

**Comment by ambv** [comment]
This looks ready for inclusion. However, since 3.10 is out and 3.9 is pretty late in the game, I'd like to learn if there is an exception to "no new features in bugfix releases" when it comes to adding MIME types.

**Comment by merwok** [comment]
I think there is an informal exception for registries that have definitions external to Python; in previous tickets it was deemed safe and beneficial to backport new entries.  I think I remember than @terryjreedy or maybe @bitdancer were involved.

Tickets with backports:
- https://bugs.python.org/issue40626
- https://bugs.python.org/issue37305
- https://bugs.python.org/issue30824
- https://bugs.python.org/issue16329
- https://bugs.python.org/issue16316 (also mentions html.entities and webbrowser: https://bugs.python.org/issue16316#msg187373)
- https://bugs.python.org/issue5301

Backport avoided to avoid compat issue: https://bugs.python.org/issue22589

→ Found the discussion of the exception!  https://bugs.python.org/issue13952#msg152752

**Comment by terryjreedy** [comment]
R. David Murray was *much* more involved in such decisions.  As the last link indicates, the debate tended to be framed as whether the item missing is a bug, rather than an exceptional new feature.

**Comment by ambv** [comment]
Given R. David's conservative approach (which I share) demonstrated in the last link by Eric, I'll merge this to 3.11 alone. Unless a backwards incompatibility is demonstrated, we won't be backporting to 3.10 and 3.9.

**Pull request merged**